packages/template-retail-react-app/app/pages/cart/index.jsx

```diff
@@ -585,23 +661,155 @@ const Cart = () => {
+                    deliveryShipments.push(shipmentData)
                 }
-                return acc
-            },
-            {regularProducts: [], bonusProducts: []}
+            }
+        })
+
+        const result = [...pickupShipments]
+
+        // Combine all delivery shipments into a single group
+        if (deliveryShipments.length > 0) {
+            const combinedDeliveryProducts = deliveryShipments.reduce(
+                (acc, shipmentData) => {
+                    acc.regularProducts.push(...shipmentData.categorizedProducts.regularProducts)
+                    acc.bonusProducts.push(...shipmentData.categorizedProducts.bonusProducts)
+                    return acc
+                },
+                {regularProducts: [], bonusProducts: []}
+            )
+
+            result.push({
+                shipment: null, // No specific shipment for combined delivery
+                isPickupOrder: false,
+                store: null, // No specific store for combined delivery
+                categorizedProducts: combinedDeliveryProducts,
+                itemsInShipment:
+                    combinedDeliveryProducts.regularProducts.length +
+                    combinedDeliveryProducts.bonusProducts.length
+            })
+        }
+
+        return result
+    }, [basket?.shipments, basket?.productItems, storeData])
+
+    /***************************** Delivery Options **************************/
+
+    const onDeliveryOptionChange = async (productItem, selectedDeliveryOption) => {
+        try {
+            setCartItemLoading(true)
+            setSelectedItem(productItem)
+
+            const selectedPickup = selectedDeliveryOption === DELIVERY_OPTIONS.PICKUP
+
+            // If the user selects pickup and no store is selected, open the store locator modal
+            if (selectedPickup && !selectedStore) {
+                storeLocatorModal.onOpen()
+                return
+            }
+
+            // Get default inventory ID from product data - throw error if not available
+            const productData = products?.[productItem.productId]
+            const defaultInventoryId = productData?.inventory?.id
+
+            if (!defaultInventoryId) {
+                throw new Error(`No inventory ID found for product ${productItem.productId}`)
+            }
+
+            await updateDeliveryOption(
+                productItem,
+                selectedPickup,
+                selectedStore,
+                defaultInventoryId
+            )
+        } catch (error) {
+            console.error('Error changing delivery option:', error)
+            showError()
+        } finally {
+            setCartItemLoading(false)
+            setSelectedItem(undefined)
+        }
+    }
+
+    // Function to render deliveryActions
+    const renderDeliveryActions = (productItem, shipmentInfo) => {
+        const showDeliveryOptions = storeLocatorEnabled && multishipEnabled
```

Comment:
Should storeLocatorEnabled be enough? since storeLocatorEnabled has takan `multishipEnabled` into account 🤔 
```suggestion
        const showDeliveryOptions = storeLocatorEnabled && multishipEnabled
```

Reply:
> Should storeLocatorEnabled be enough?

no this feature requires both store locator and multiship to be enabled (remember they are independent flags)

Reply:
We'll need both here since we only show if both BOPIS and Multiship are enabled, one can be disabled/enabled separately from the other.

Reply:
oh right, gotcha. I thought it was storeLocatoDisabled bool in other file 🫠